examples/seq2seq/utils_seq2seq.py

```diff
@@ -0,0 +1,226 @@
+"""
+This defines an input example format as well as classes to hold the data.
+The logic here was largely derived from that in utils_ner.py though a lot of if statements for tokenization
+was moved into the convert_examples_into_features function.
+"""
+import logging
+import os
+
+
+logger = logging.getLogger(__name__)
+
+
+def pairwise(it):
+    """
+    A function to pairwise retrieve pieces of an iterable
+    If given [1,2,3,4], this will return
+    (1,2)
+    (3,4)
+    """
+    it = iter(it)
+    try:
+        while True:
+            yield next(it), next(it)
+    # in python3.7 this raises a stopiteration that blocks example reading
+    except StopIteration:
+        pass
+
+
+class InputExample(object):
+    """A single training/test example for token classification."""
+
+    def __init__(self, guid, words, formatted_words=None):
+        """Constructs a InputExample.
+
+        Args:
+            guid: Unique id for the example.
+            words: list. The words of the sequence.
+            formatted_words: (Optional) list. The formatted version of the input words. This should be
+            specified for train and dev examples, but not for test examples.
+        """
+        self.guid = guid
+        self.words = words
+        self.formatted_words = formatted_words
+
+
+"""
+Example structure:
+input:  march 23 2019
+output: 3/23/2019
+"""
+
+
+def read_examples_from_file(data_dir, mode):
+    file_path = os.path.join(data_dir, "{}.txt".format(mode))
+    guid_index = 1
+    examples = []
+    with open(file_path, encoding="utf-8") as f:
+        words = []
+        formatted_words = []
+        for words, formatted_words in pairwise(f):
+            words = words.lstrip("input: ").strip()
+            formatted_words = formatted_words.lstrip("output: ").strip()
+            examples.append(
+                InputExample(guid="{}-{}".format(mode, guid_index), words=words, formatted_words=formatted_words,)
+            )
+            guid_index += 1
+    return examples
+
+
+class InputFeatures(object):
+    """A single set of features of data."""
+
+    def __init__(
+        self, input_ids, output_ids, input_mask, output_mask, segment_ids, formatted_tokens,
+    ):
+        self.input_ids = input_ids
+        self.output_ids = output_ids
+        self.input_mask = input_mask
+        self.output_mask = output_mask
+        self.segment_ids = segment_ids
+        self.formatted_tokens = formatted_tokens
+
+
+def convert_examples_to_features(
+    examples, max_seq_length, tokenizer, model_type, pad_token_label_id=-100,
+):
+    """
+    Loads a data file into a list of `InputBatch`s
+    """
+    #  `cls_token_segment_id` define the segment id associated to the CLS token (0 for BERT, 2 for XLNet)
+    cls_token_segment_id = 2 if model_type in ["xlnet"] else 0
+    pad_token_segment_id = 4 if model_type in ["xlnet"] else 0
+    features = []
+
+    cls_token, sep_token, pad_token = cls_token = (
+        tokenizer.cls_token,
```

Comment:
Hi, I've been using your example to guide my own setup of a seq2seq model (thank you!) Is this line in your pull request a typo? Think it's just supposed to be cls_token, sep_token, pad_token = ( ...